src/javascript/app_2/pages/trading/components/elements/tabs.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+
+class Tabs extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = { active: '1' };
+    }
+    render() {
+        const className = 'tab-container';
+        return (
+            <div className={`${className} ${this.props.alignment}`}>
+                <TabsWrapper
+                  active={this.state.active}
+                  onChange={active => this.setState({active})}
+                >
+                {Object.keys(this.props.list).map(key => (
+                    <React.Fragment key={key}>
+                        {!!this.props.list[key].icon &&
+                            <i className={this.props.list[key].icon}></i>
+                        }
+                        <span>{this.props.list[key].header}</span>
+                    </React.Fragment>
+                ))}
+                </TabsWrapper>
+                <p className='tab-content'>{this.props.list[this.state.active].content}</p>
+            </div>
+        );
+    }
+};
+
+class TabsWrapper extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            sizes: {},
+        };
+        this.els = {};
+    }
+
+    componentDidMount() {
+        this.getSizes();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.children !== this.props.children && prevProps.active !== this.props.active) {
+            this.getSizes();
+        }
+    }
+
+    getSizes() {
+        const rootBounds = this.root.getBoundingClientRect();
+
+        const sizes = {};
+
+        Object.keys(this.els).forEach((key) => {
+            const el = this.els[key];
+            const bounds = el.getBoundingClientRect();
+
+            const left = bounds.left - rootBounds.left;
+            const right = rootBounds.right - bounds.right;
+
+            sizes[key] = {left, right};
+        });
+
+        this.setState({sizes});
+        return sizes;
+    }
+
+    render() {
+        // console.log(JSON.stringify(this.state, null, 2));
+        return (
+          <div
+            className='tab-wrapper'
+            ref={el => this.root = el}
+          >
+            {React.Children.map(this.props.children, (child, i) => {
+                const className = 'tab';
+                return (
+                  <React.Fragment key={i}>
+                      <div
```

Comment:
We don't need the `Fragement`, add the `key` to `div` ... like `<div key={i} ...`